example/convolution1D/src/convolution1D.cpp

```diff
@@ -0,0 +1,200 @@
+/* Copyright 2023 Benjamin Worpitz, Erik Zenker, Bernhard Manfred Gruber, Jan Stephan
+ * SPDX-License-Identifier: ISC
+ */
+
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <type_traits>
+//! Convolution Example
+//!
+//! 1D convolution example: Creates two 1D arrays, calculates the convolution integral using those arrays.
+//! Array sizes are hardcoded.
+//!
+
+// Size of 1D arrays to be used in convolution integral
+// In signal processing domain, the term "kernel" is used for the matrix or array used in convolution integral.
+// Here instead of "kernel" the term "filter" is used because kernel has a different meaning in GPU programming.
+constexpr size_t FilterSize = 3;
+constexpr size_t InputSize = 8;
+
+constexpr float ExpectedOutput[InputSize] = {0.8f, 1.4f, 2.0f, 2.6f, 3.2f, 3.8f, 4.4f, 2.3f};
+
+/**
+ * @brief The ConvolutionKernel function object
+ * Calculates 1D convolution integral using input and filter arrays.
+ */
+struct ConvolutionKernel
+{
+    /** @brief Main convolution code
+     *  @param Accelerator
+     *  @param Input array, first input of convolution integral
+     *  @param Filter array, second input of convolution integral
+     *  @param Empty output array to be filled
+     *  @param Input array size
+     *  @param Filter size
+     */
+    template<typename TAcc, typename TElem>
+    ALPAKA_FN_ACC auto operator()(
+        TAcc const& acc,
+        TElem const* const input,
+        TElem const* const filter,
+        TElem* output,
```

Comment:
```suggestion
        TElem* const output,
```

Reply:
Ok the pointer is const.

Reply:
Year, this is a confusing part of C :grimacing:

Reply:
done thanks.